### PR TITLE
docs: Added documentation for device twin module

### PR DESCRIPTION
**Pull request by sujithsimon22**
**What type of PR is this?**

 kind documentation

**What this PR does / why we need it**:

This PR contains the documentation of device twin module which will help users and developers in  understanding the module and its functionalities better.

### CODE REVIEW

File: docs/modules/devicetwin.md [hunk 1]
```diff
@@ -0,0 +1,328 @@
+
+# DeviceTwin
+
+
+## Overview
+
+DeviceTwin is responsible for storing device status and syncing device status to the cloud. 
+It also provides query interfaces for applications.
```
Comment (rohitsardesai83) [inline docs/modules/devicetwin.md]: what kind of query interface ? can we have a link to a section below which describes these interfaces ?

File: docs/modules/devicetwin.md [hunk 2]
```diff
@@ -0,0 +1,328 @@
+
+# DeviceTwin
+
+
+## Overview
+
+DeviceTwin is responsible for storing device status and syncing device status to the cloud.
```
Comment (rohitsardesai83) [inline docs/modules/devicetwin.md]: dealing with device status in one of the jobs of device twin. reading later through the document, we can see it also deals with device twin attributes, membership events etc. Can we describe all of the responsibilities of device twin in a summary?

File: docs/modules/devicetwin.md (hunk 1 above)
Comment (rohitsardesai83) [reply docs/modules/devicetwin.md]: we can put a block diagram which depicts what are the submodules of device twin . each sub module's job can be expanded in later sections with more diagrams/images. we can do this in next PR incrementally.

File: docs/modules/devicetwin.md [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+
+DeviceTwin is responsible for storing device status and syncing device status to the cloud. 
+It also provides query interfaces for applications.
+   
+   
+## Operations Performed By Device Twin Controller
+ 
+ The following are the functions performed by device twin controller :-
+  
+  - Sync Sql Lite
```
Comment (rohitsardesai83) [inline docs/modules/devicetwin.md]: can we rephrase this to "Sync metadata to/from db ( Sqlite )" . in the code it might be with this method name. but in the docs we can make it more clear.

File: docs/modules/devicetwin.md [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+  - Register and Start Modules
```
Comment (rohitsardesai83) [inline docs/modules/devicetwin.md]: before this , in the overview we can mention about the sub-modules and here we can say "Register and start sub-modules"

File: docs/modules/devicetwin.md [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+  - Distribute message
```
Comment (rohitsardesai83) [inline docs/modules/devicetwin.md]: Distribute messages to ? sub-modules ? Can we make it more clear ?

File: docs/modules/devicetwin.md (hunk 1 above)
Comment (sujithsimon22) [reply docs/modules/devicetwin.md]: sure @rohitsardesai83 , In the subsequent PRs we will incrementally add diagrams, images or flowcharts in order to make it easier for developers to understand the device twin module.

File: docs/modules/devicetwin.md [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+  
+### Health Check 
+ 
+ Health monitoring is done by sending a ping message to all device twin modules every minute and receiving it from the module.
+ If the time contained in the health message received from the module is more than two minutes old, then the module is restarted.
+  
+
+## Modules
+
+DeviceTwin consists of four modules, namely :-
```
[the remark this answers is not in the document]
Comment (sujithsimon22) [reply docs/modules/devicetwin.md]: will be handled in the next PR

File: docs/modules/devicetwin.md [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+
+**dealSendToEdge**:  dealSendToEdge() is used to send data to the other modules present at the edge.
+                     This function sends the message received to the edgeHub module using beehive framework.
+                     The edgeHub module after receiving the message will send it to the required recipient. 
+                   
+**dealLifeCycle**:   dealLifeCycle() checks if the cloud is connected and the state of the twin is disconnected, it then changes the status 
+                     to connected and sends the node details to edgehub. If the cloud is disconnected then, it sets the state of the twin
+                     as disconnected. 
+
+**dealConfirm**:     dealConfirm() is used to confirm the event. It checks whether the type of the message is right and
```
[the remark this answers is not in the document]
Comment (sujithsimon22) [reply docs/modules/devicetwin.md]: This will be target in the next PR

**Comment by m1093782566** [comment]
@sujithsimon22 @rohitsardesai83 

We still miss edged module detailed document?

**Comment by sameersarkar1979** [comment]
@sujithsimon22 @m1093782566 

I will take care of edged documentation.
